dolphin-dnn/src/main/java/edu/snu/dolphin/dnn/layerparam/provider/LocalNeuralNetParameterProvider.java

```diff
@@ -43,21 +43,28 @@
   private final LayerParameter[] deltaLayerParameters;
   private final float stepsize;
   private int numUpdate = 0;
+  private final MatrixFactory matrixFactory;
 
   @Inject
   public LocalNeuralNetParameterProvider(
-      @Parameter(SerializedLayerConfigurationSet.class) final Set<String> serializedLayerConfigurationSet,
+      @Parameter(SerializedLayerConfigurationSet.class)final Set<String> serializedLayerConfigurationSet,
       @Parameter(Stepsize.class) final float stepsize,
-      final ConfigurationSerializer configurationSerializer) {
+      final ConfigurationSerializer configurationSerializer,
+      final MatrixFactory matrixFactory) {
     this.layerParameters = new LayerParameter[serializedLayerConfigurationSet.size()];
     this.deltaLayerParameters = new LayerParameter[serializedLayerConfigurationSet.size()];
     this.stepsize = stepsize;
+    this.matrixFactory = matrixFactory;
+
+    final Configuration matrixFactoryConf = Tang.Factory.getTang().newConfigurationBuilder()
+        .bindImplementation(MatrixFactory.class, matrixFactory.getClass())
+        .build();
 
     for (final String serializedInitializerConfiguration : serializedLayerConfigurationSet) {
       try {
         final Configuration initializerConfiguration =
             configurationSerializer.fromString(serializedInitializerConfiguration);
-        final Injector injector = Tang.Factory.getTang().newInjector(initializerConfiguration);
+        final Injector injector = Tang.Factory.getTang().newInjector(initializerConfiguration, matrixFactoryConf);
```

Comment:
Can we receive an Injector at the constructor and call `injector.forkInjector()` here? That way, we don't have to create `matrixFactoryConf`.